x/slashing/keeper/infractions.go

```diff
@@ -137,11 +137,7 @@ func (k Keeper) HandleValidatorSignatureWithParams(ctx context.Context, params t
 	// if we are past the minimum height and the validator has missed too many blocks, punish them
 	if height > minHeight && signInfo.MissedBlocksCounter > maxMissed {
 		modifiedSignInfo = true
-		validator, err := k.sk.ValidatorByConsAddr(ctx, consAddr)
-		if err != nil {
-			return err
-		}
-		if validator != nil && !validator.IsJailed() {
+		if val != nil && !val.IsJailed() {
```

Comment:
hmmm do we  need to check here, IsJailed is already checked above and if the val is nil then probably it already return an err when we get it

Reply:
this is true, this function could probably be cleaned up